apps/editing-toolkit/editing-toolkit-plugin/error-reporting/index.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Wordpress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * External dependences
+ */
+import { throttle } from 'lodash';
+
+/**
+ * Interval for error reports so we don't flood te endpoint.
+ *
+ * @type {number} throttling interval (since the last request) in milliseconds.
+ */
+const REPORT_INTERVAL = 60000; // 1 minute
+
+/**
+ * Errors that happened before this script had a chance to load
+ * are captured in a global array. See `./index.php`.
+ */
+const headErrors = window._jsErr || [];
+const headErrorHandler = window._headJsErrorHandler;
+
+const reportError = throttle( ( { error } ) => {
+	console.log( 'reporting error!', error.message );
+	const data = {
+		message: error.message,
```

Comment:
This will break for cors error events because they don't have an `Error` object instance. See: p1612923248252300-slack-C7YPUHBB2.

Reply:
The actual error event for cors errors is pretty useless, it doesn't even contain the lineno and column, just the file name. I guess it's better than nothing though and we should still capture them.